specutils/spectra/spectrum1d.py

```diff
@@ -442,6 +442,81 @@ def _spectral_slice(self, item):
         reg = SpectralRegion(start, stop)
         return extract_region(self, reg)
 
+    def collapse(self, method, axis=None, physical_type=None):
+        """
+        Collapse the flux array given a method. Will collapse either to a single
+        value (default), over a specified numerical axis or axes if specified, or
+        over the spectral or non-spectral axes if ``physical_type`` is specified.
+
+        If the collapse leaves the spectral axis unchanged, a `~specutils.Spectrum1D`
+        will be returned. Otherwise an `~astropy.units.Quantity` array will be
+        returned.
+
+        Note that these calculations are not currently uncertainty-aware, but do
+        respect masks.
+
+        Parameters
+        ----------
+
+        method : str
+            The method by which the flux will be collapsed. Options are 'mean',
+            'min', 'max', 'sum', and 'median'.
+        axis : int, tuple, optional
+            The axis or axes over which to collapse the flux array.
+        physical_type : str, optional
+            Either 'spectral' or 'spatial'.
+
+        Returns
+        -------
+        :class:`~specutils.Spectrum1D` or :class:`~astropy.units.Quantity`
+
+        """
+        collapse_func = {"mean": np.nanmean, "max": np.nanmax, "min": np.nanmin,
+                         "median": np.nanmedian, "sum": np.nansum}
+
+        if physical_type is not None and axis is not None:
+            raise ValueError("Cannot set both axis and physical_type inputs")
+
+        if physical_type == 'spectral':
+            axis = -1
+        elif physical_type == 'spatial':
+            # generate tuple if needed for multiple spatial axes
+            axis = tuple([x for x in range(len(self.flux.shape) - 1)])
+        else:
+            if physical_type is not None:
+                raise ValueError("physical_type must be 'spatial' or 'spectral'")
+
+        # Set masked locations to NaN for the calculation, since the `where` argument
+        # does not seem to work consistently in the numpy functions.
+        flux_to_collapse = self.flux.copy()
+        flux_to_collapse[np.where(self.mask != 0)] = np.nan
```

Comment:
Isn't the mask already a boolean array? Does simply using `flux_to_collapse[~self.mask] = np.nan` not work?